pkg/cloud/gateway/gateway.go

```diff
@@ -703,15 +703,20 @@ func (s *LocalGatewayService) Start(opts *gateway.GatewayStartOpts) error {
 	return s.serviceServer.Serve(s.serviceListener)
 }
 
-func (s *LocalGatewayService) Stop() error {
-	for _, s := range s.apiServers {
-		// shutdown all the servers
-		// this will allow Start to exit
+func shutdownServers(servers []*apiServer) {
+	for _, s := range servers {
+		// Shutdown all the servers
+		// This will allow Start to exit
 		ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond*100)
 		defer cancel()
 
 		_ = s.srv.ShutdownWithContext(ctx)
 	}
+}
+
+func (s *LocalGatewayService) Stop() error {
+	shutdownServers(s.apiServers)
+	shutdownServers(s.httpServers)
```

Comment:
Does this need to be applied to websocket servers as well?

Reply:
added that too